ext/opcache/Optimizer/pass1_5.c

```diff
@@ -309,6 +307,11 @@ void zend_optimizer_pass1(zend_op_array *op_array, zend_optimizer_ctx *ctx)
 						}
 					}
 				} else if (op_array->scope &&
+					ZEND_OP1_TYPE(opline) == IS_UNUSED &&
+					(opline->op1.num & ~ZEND_FETCH_CLASS_NO_AUTOLOAD) == ZEND_FETCH_CLASS_SELF) {
```

Comment:
The whole next block (L314-L321) may be removed.
I see no problems with this block (L309-L313)
Do I understand you correctly?